### ISSUE TITLE
Is CHW format mandatory in inference ?

### ISSUE THREAD

**Issue by andreidi** [parallel-forall/code-samples#10]
So, the model was loaded/trained with Keras/Tensorflow and for sure input is HWC however during inference I see both in testing and conversion scrips we are forcing CHW input. Why is that and how is that possible?

**Comment by faustomilletari** [comment]
There is conversion going on. You have your tensor flow model in hwc and once it's in tensor rt everything will be chw.



Fausto Milletarì

**Comment by andreidi** [comment]
In understand. Does this mean when input tensor if 4D the convertor automatically assumes swapping of dim 2 and dim 4 (certainly for the case when input is tensorflow frozen graphs I mean) ?
